slam_toolbox/src/merge_maps_kinematic.cpp

```diff
@@ -267,6 +286,10 @@ bool MergeMapsKinematic::mergeMapCallback(
   map.map.header.frame_id = "map";
   sstS_[0].publish(map.map);
   sstmS_[0].publish(map.map.info);
+
+  merged_mapper -> GetGraph()->CorrectPoses();
+  std::string filename = req.filename;
```

Comment:
You need to do some error checking on this. Making sure that filename is valid or filled in. Or fail or default to some string with a warning. 

finally, you need to expose this to the rviz plugin so that the filename can be populated.

Reply:
hello, done